evm/db/backends/level.py

```diff
@@ -0,0 +1,48 @@
+import shutil
+from .snapshot import Snapshot
+from .base import (
+    BaseDB,
+)
+
+
+class LevelDB(BaseDB):
+    db = None
+
+    # Creates db as a class variable to avoid level db lock error
+    def __init__(self):
+        try:
+            import leveldb
+        except ImportError:
+            raise ImportError("LevelDB requires the leveldb \
+                               library which is not available for import.")
+        if not LevelDB.db:
+            LevelDB.db = leveldb.LevelDB('./evm/db/backends/db',
+                                         create_if_missing=True, error_if_exists=False)
+
+    def get(self, key):
+        return bytes(self.db.Get(key))
+
+    def set(self, key, value):
+        self.db.Put(key, value)
+
+    def exists(self, key):
+        return key in list(self.db.RangeIter(include_value=False))
```

Comment:
This strikes me as something that we can't do unless we have a way to lookup key existence in a way that is O(log(n)) or less. This is currently O(n).